Frameworks/UIKit/UIImage.mm

```diff
@@ -671,38 +667,45 @@ - (void)draw1PartImageInRect:(CGRect)pos {
  @Status Interoperable
 */
 - (void)drawAtPoint:(CGPoint)point {
-    [self drawAtPoint:CGPointMake(point.x, point.y) blendMode:kCGBlendModeNormal alpha:0.0f];
+    [self drawAtPoint:point blendMode:kCGBlendModeNormal alpha:1.0f];
 }
 
 /**
  @Status Interoperable
 */
-- (void)drawAtPoint:(CGPoint)point blendMode:(CGBlendMode)mode alpha:(float)alpha {
+-(void)drawAtPoint:(CGPoint)point blendMode : (CGBlendMode)mode alpha : (float)alpha {
     CGContextRef cur = UIGraphicsGetCurrentContext();
+    CGImageRef img = getImage(self);
+
     if (!cur) {
         TraceVerbose(TAG, L"CGContext = NULL!");
         return;
     }
-    if (!getImage(self)) {
+
+    if (!img) {
         TraceVerbose(TAG, L"m_pImage = NULL!");
         return;
     }
 
     CGContextSaveGState(cur);
+
     CGContextSetBlendMode(cur, mode);
+    CGContextSetAlpha(cur, alpha);
+
+    CGImageBacking* imgBacking = img->Backing();
 
     CGRect srcRect;
     CGRect pos;
     pos.origin = point;
-    pos.size.width = ((float)getImage(self)->Backing()->Width() / _scale);
-    pos.size.height = ((float)getImage(self)->Backing()->Height() / _scale);
+    pos.size.width = ((float)imgBacking->Width());
+    pos.size.height = ((float)imgBacking->Height());
 
     srcRect.origin.x = 0;
-    srcRect.origin.y = float(getImage(self)->Backing()->Height());
-    srcRect.size.width = float(getImage(self)->Backing()->Width());
-    srcRect.size.height = -((float)getImage(self)->Backing()->Height());
+    srcRect.origin.y = float(imgBacking->Height());
```

Comment:
>float( [](start = 23, length = 6)

nit: it's a bid odd that two of these are constructing a float rather than casting to a float